core_backend/app/question_answer/models.py

```diff
@@ -204,90 +205,35 @@ async def save_query_response_to_db(
         The user query response database object.
     """
 
-    user_query_responses_db = QueryResponseDB(
-        query_id=user_query_db.query_id,
-        search_results=response.model_dump()["search_results"],
-        llm_response=response.model_dump()["llm_response"],
-        response_datetime_utc=datetime.now(timezone.utc),
-    )
+    if type(response) is QueryResponse:
+        user_query_responses_db = QueryResponseDB(
+            query_id=user_query_db.query_id,
+            search_results=response.model_dump()["search_results"],
+            llm_response=response.model_dump()["llm_response"],
+            response_datetime_utc=datetime.now(timezone.utc),
+            debug_info=response.model_dump()["debug_info"],
+            is_error=False,
+        )
+    elif type(response) is QueryResponseError:
```

Comment:
So we are keeping two schemas but just one db model. Would it make sense to consolidate schemas as well?

Reply:
Yeah because QueryResponseError is for when we return an error 400 (guardrail failure) and has extra error info, and QueryResponse is for a successful call. We also use isinstance with QueryResponseError to skip guardrails if previous ones have failed.